.github/workflows/release-testpypi.yml

```diff
@@ -0,0 +1,38 @@
+---
+name: publish wheels to testpypi
+
+on:
+  push:
+    branches:
+      - main
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+
+    environment:
+      name: testpypi
+      url: https://test.pypi.org/p/python-gilt
+
+    permissions:
+      # mandatory for trusted publishing
+      id-token: write
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - run: git fetch --force --tags
+      - uses: actions/setup-go@v5
+        with:
+          go-version: stable
+      - uses: goreleaser/goreleaser-action@v5
+        with:
+          distribution: goreleaser
+          version: latest
+          args: release --clean --snapshot
```

Comment:
`--clean` removes `dist/` prior to execution, and leaves files in place for subsequent step.

Reply:
Using `--snapshot` as we likely want to publish test packages with `next-tag`.dev?

Reply:
@0xDEC0DE is there a better pythonism for .dev packages?

`goreleaser` is generating the packge name as ` version=2.0.4.dev` as it's likely looking at the next available tag.  For pushes to testpypi, we probably want a dev version to include the git hash as part of the version?

Reply:
We can also skip testpypi all together, it just seemed somewhat useful.

Reply:
I've usually considered pre-release packages to be more trouble than they are worth.

Reply:
It deff seems like it :)